esphome/components/thermostat/thermostat_climate.cpp

```diff
@@ -1330,45 +1330,64 @@ void ThermostatClimate::set_heat_deadband(float deadband) { this->heating_deadba
 void ThermostatClimate::set_heat_overrun(float overrun) { this->heating_overrun_ = overrun; }
 void ThermostatClimate::set_supplemental_cool_delta(float delta) { this->supplemental_cool_delta_ = delta; }
 void ThermostatClimate::set_supplemental_heat_delta(float delta) { this->supplemental_heat_delta_ = delta; }
+
+void ThermostatClimate::set_timer_duration_in_sec_(ThermostatClimateTimerIndex timer_index, uint32_t time) {
+  uint32_t new_duration_ms = 1000 * (time < this->min_timer_duration_ ? this->min_timer_duration_ : time);
+
+  if (this->timer_[timer_index].active) {
+    // Timer is running, calculate elapsed time and adjust if needed
+    uint32_t current_time = millis();
```

Comment:
For consistency, I think this should use `App.get_loop_component_start_time()`:

```suggestion
    uint32_t current_time = App.get_loop_component_start_time();
```

Reply:
Should I also change this on line 900 where the timer is started? That's somewhat out of scope for this PR, is the same inconsistency.

Reply:
I believe the use of `millis()` on line 900 was because `start_timer_` is called in `setup()` -- initially there was some issue there which I seem to recall has since been fixed. Regardless, we're trying to avoid adding new calls to `millis()` as it's slower and bigger than calling `App.get_loop_component_start_time()`. 🙂